File: AkitaModel/SmartScan/Events/SlotStatus.cs
namespace Skyware.Lis.AkitaModel.SmartScan.Events;


/// <summary>
/// Represents slot for ad hoc client for taking pictures.
/// This is an event payload object (SignalR).
/// </summary>
public class SlotStatus : ITypedEvent
{

    /// <summary>
    /// Id of a slot (encoded in endpoint URL).
    /// </summary>
    public string Id { get; set; }

    /// <summary>
    /// True if slot is available.
    /// </summary>
    public bool IsValid { get; set; } = false;

    /// <inheritdoc/>
    public string MethodName => "slot";

}

File: AkitaMotelTS/src/SmartScan/Events/SlotStatus.ts
/**
 * Skyware.Lis.AkitaModel.SmartScan.Events
 */

import { ITypedEvent } from './ITypedEvent';

/**
 * Represents slot for ad hoc client for taking pictures.
 * This is an event payload object (SignalR).
 */
export class SlotStatus implements ITypedEvent {
  /**
   * Id of a slot (encoded in endpoint URL).
   */
  id?: string;

  /**
   * True if slot is available.
   */
  isValid: boolean = false;

  /** @inheritdoc */
  get methodName(): string {
    return 'slot';
  }
}
